### PR TITLE
[IMP] survey: improve way to compute the list of questions

### PR DESCRIPTION
**Pull request by JKE-be**
Old code take ~5s of python, now less than 0.5s for a certification on our prod.

### CODE REVIEW

**Comment by JKE-be** [comment]
@robodoo delegate=pivi-odoo

//cc @tde-banana-odoo la PR discutée ce midi pour survey

**Review by pivi-odoo — APPROVED** [review]
LGTM, confirmed effectiveness/speedup of the patch.

**Review by tde-banana-odoo — APPROVED** [review]
Indeed, old computation was strange. I think at first implementation is_page was not stored, which is probably why computation relied only on indexes ...

File: addons/survey/models/survey_question.py [hunk 1]
```diff
@@ -264,15 +264,8 @@ def _compute_question_ids(self):
...
-                next_page_index = False
-                for page in question.survey_id.page_ids:
-                    if page._index() > question._index():
-                        next_page_index = page._index()
-                        break
-
-                question.question_ids = question.survey_id.question_ids.filtered(
-                    lambda q: q._index() > question._index() and (not next_page_index or q._index() < next_page_index)
-                )
+                question.question_ids = question.survey_id.question_ids\
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_question.py:267]: Note that method docstring is somehow outdated (and can probably be simply removed)

Comment (JKE-be) [reply addons/survey/models/survey_question.py:265]: bien vu ;) updated

**Review by tde-banana-odoo — APPROVED** [review]

**Comment by pivi-odoo** [comment]
@robodoo r+

**Pull request closed without merging**